x/staking/keeper/keeper_test.go

```diff
@@ -112,3 +115,33 @@ func (s *KeeperTestSuite) TestLastTotalPower() {
 func TestKeeperTestSuite(t *testing.T) {
 	suite.Run(t, new(KeeperTestSuite))
 }
+
+func (s *KeeperTestSuite) TestDiffCollsMigration() {
+	s.SetupTest()
+
+	delAddrs, valAddrs := createValAddrs(100)
+	err := testutil.DiffCollectionsMigration(
+		s.ctx,
+		s.key,
+		100,
+		func(i int64) {
+			ubd := stakingtypes.UnbondingDelegation{
+				DelegatorAddress: delAddrs[i].String(),
+				ValidatorAddress: valAddrs[i].String(),
+				Entries: []stakingtypes.UnbondingDelegationEntry{
+					{
+						CreationHeight: i,
+						CompletionTime: time.Unix(i, 0).UTC(),
+						Balance:        math.NewInt(i),
+						UnbondingId:    uint64(i),
+					},
+				},
+			}
+			err := s.stakingKeeper.SetUnbondingDelegation(s.ctx, ubd)
+			s.Require().NoError(err)
+		},
+		"d03ca412f3f6849b5148a2ca49ac2555f65f90b7fab6a289575ed337f15c0f4b",
```

Comment:
out of curiosity, how was this hash obtained?
If we want to remove all doubt a pre-collections migration test should be pushed which just runs this testutil fn, asserting the current state.  In the migration PR we leave the test unchanged.  That we *know* for certain that the original hash was obtained in the pre-collections code path.

feel free to tell me I'm being too cautious here.

Reply:
> out of curiosity, how was this hash obtained? If we want to remove all doubt a pre-collections migration test should be pushed which just runs this testutil fn, asserting the current state. In the migration PR we leave the test unchanged. Then we _know_ for certain that the original hash was obtained in the pre-collections code path.
> 
> feel free to tell me I'm being too cautious here.

This hash is obtained by running this test on main and we make sure the same hash is returned when tested on feature branch. 
I definitely agree that this might seem uncertain not knowing if the hash is coming from main or not. Having pre-collections migration test would make things clear here. But I'm not sure if we could do that in this PR. Maybe I'll ask @testinginprod  or @facundomedica  and raise a separate PR for these pre-collections migration tests. Anyways, we can always clear up this unwanted code later when all the state objects migrated to collections.

Reply:
> sorry, I'm catching up on this collections work. It looks like [SetUnbondingDelegation](https://github.com/cosmos/cosmos-sdk/blob/09db6997c532e984905b33f8618b57b2bbd1fed1/x/staking/keeper/delegation.go#L309-L328) is presently outside of collections and therefore already fulfills what I'm proposing above ^^. I guess we're planning to migrate that fn at some point in x/staking migration work?

yeah, the full func migration does happen with work from #17270. All these PRs are meant be simpler with migration of single state object at a time.

Reply:
I think it's ok to trust that the hash is ok, should be fairly easy to get it and check it's the same. But I also think it's ok to doubt about it, maybe we can do something in the migrations package? So we have the same function in the main code and in the migrations package. So we make sure that the hash matches between those two tests with no need to go back to main. Although you'd need to get it from main in the beginning anyway.
But yeah, I would keep that out of this PR for now